src/dfcx_scrapi/tools/test_cases_util.py

```diff
@@ -0,0 +1,559 @@
+
+        intent_id = intents_map.get(intent_name, None)
+
+        return intent_id
+
+    def _get_intent_name_by_intent_id(
+        self,
+        intent_id: str,
+        intents_map: Dict[str, str]) -> str:
+        """Attempt to get the intent display name from the Intents Map
+            by the intent's display name."""
+
+        intent_name = intents_map.get(intent_id, None)
+
+        return intent_name
+
+    def _get_intent_id_from_virtual_agent_output(
+        self,
+        virtual_agent_output: types.ConversationTurn.VirtualAgentOutput,
+        target: dict) -> Union[str, None]:
+        """Attempt to find the intent id from the source's agent 
+        conversation_turn.virtual_agent_output triggered intent
+        display name using the target's intents map
+        """
+
+        source_intent = (
+          virtual_agent_output.triggered_intent.display_name
+        )
+        target_intent_id = self._get_intent_id_by_intent_name(
+          intent_name = source_intent,
+          intents_map = target["intents_map_reverse"])
+
+        return target_intent_id
+
+    def _get_page_id_from_virtual_agent_output(
+        self,
+        virtual_agent_output: types.ConversationTurn.VirtualAgentOutput,
+        source: dict,
+        target: dict) -> Union[str, None]:
+        """Attempt to find the page id from the source's agent conversation turn
+          virtual agent output page id to the target's pages map
+        """
+
+        source_page = virtual_agent_output.current_page.display_name
+        source_page_id = virtual_agent_output.current_page.name
+        source_flow = self._get_flow_display_name_by_id(
+          flow_id = source_page_id.split("/pages/")[0],
+          flows_map = source["flows_map"])
+        target_flow_id = self._get_flow_id_by_flow_name(
+          flow_name = source_flow,
+          flows_map = target["flows_map_reverse"])
+        target_page_id = self._get_page_id_by_page_name(
+          flow_id = target_flow_id,
+          page_name = source_page,
+          pages_map = target["pages_map_reverse"])
+
+        return target_page_id
+
+    def _get_intent_id_from_user_input(
+        self,
+        user_input: types.ConversationTurn.UserInput,
+        source: dict,
+        target: dict) -> Union[str, None]:
+        """Attempt to find the intent id from the source's agent 
+          conversation turn user input intent id to the target's pages map
+        """
+
+        source_intent_id = user_input.input.intent
+        source_intent = self._get_intent_name_by_intent_id(
+            intent_id = source_intent_id,
+            intents_map = source["intents_map"])
+        target_intent_id = self._get_intent_id_by_intent_name(
+            intent_name = source_intent,
+            intents_map = target["intents_map_reverse"])
+
+        return target_intent_id
+
+    def _get_new_conversation_turns(
+        self,
+        source: dict,
+        target: dict,
```

Comment:
Since these two represent agent ids, use `source_agent_id` and `target_agent_id` instead

Reply:
oh they are not agent ids. they are commons_config dictionaries. Fixed the comments

Reply:
Let's update the args to reflect something more semantically meaningful like `source_resource_map`, `target_resource_map`.  

This is in conjunction with my other comment about not using the word `common` as it's quite vague.

Reply:
Also, let's update the name of this method to `_build_conversation_turns_for_target_agent` to be more explicit about what this is doing.  We're not actually getting conversation turns here.

Reply:
- Changed the name of the function to "_build_conversation_turns_for_target_agent"
- Instead of commons_config, changed the name to "resource_map"